wrapper/src/main/java/software/amazon/jdbc/ConnectionPluginManager.java

```diff
@@ -372,6 +435,7 @@ public void initHostProvider(
         () -> {
           throw new SQLException("Shouldn't be called.");
         });
+    context.closeContext();
```

Comment:
executeWithSubscribedPlugins throws the exception, so this line needs to be in a finally block.